packages/core-flows/src/definition/cart/steps/validate-cart-shipping-options.ts

```diff
@@ -0,0 +1,78 @@
+import { ModuleRegistrationName } from "@medusajs/modules-sdk"
+import { CartDTO } from "@medusajs/types"
+import { arrayDifference, MedusaError } from "@medusajs/utils"
+import { createStep, StepResponse } from "@medusajs/workflows-sdk"
+import { IFulfillmentModuleService } from "../../../../../types/dist/fulfillment/service"
+
+interface StepInput {
+  cart: CartDTO
+  option_ids: string[]
+}
+
+export const validateCartShippingOptionsStepId =
```

Comment:
Validating and refetching seems like it has a very similar logic, any chance we can somehow refactor the two?